### DISCUSSION TITLE
Add Mastodon Social

### DISCUSSION THREAD

**Discussion by jvanderaa** [mmistakes/minimal-mistakes#4044] [Ideas]
There are social links for Facebook, bird site, etc. Mastodon is taking off as the bird site has some significant shifts in direction. Would like to have Mastodon as a "first class" citizen of Minimal Mistakes.

**Comment by mwweinberg** [comment]
I've managed to get 80-ish% of the way there, so I'm posting this here just in case it helps someone else.

1. Edit _config.yml to add a "mastodon: [your name]" line in the social section
2. Edit _includes/navigation.html to add a new if statement in the list of social if statements (it was on line 44 for me):
```
{% if site.owner.mastodon %}<li>
    <a href="[your instance]/@{{ site.owner.mastodon }}"><i class="fa fa-fw fa-mastodon"></i> Mastodon</a>
</li>{% endif %}
```
3. Edit _site/assets/css/main.css to add:
```
.fa-mastodon:before {
    content: ""
}
```

What's the last 20%? If you do all of that and cut and paste the character between the quote marks in the 
`content: ""`
line from the existing fa-twitter code, it will use a twitter icon. If you cut and paste it from the facebook code, it will use the facebook icon. Since that icon just shows up as a rectangle to me, I have no idea what information is hidden in it, or how to update it to include a new icon.

**Comment by jdeluyck** [comment]
For future reference: you can just add this to your _config.yml footer section:

```
footer:
  links:
    - label: "Mastodon"
      icon: "fab fa-fw fa-mastodon"
      url: https://fediverseinstance/@profile
      rel: "me"
```

I haven't found yet how to get a "share to mastodon" button, though.
